src/hb-ot-post-table.hh

```diff
@@ -55,6 +55,91 @@ struct postV2Tail
     return_trace (glyphNameIndex.sanitize (c));
   }
 
+  void unpackPStrings (unsigned max_index, hb_map_t& index_map) const
+  {
+    if (max_index < 258) return;
+    const UnsizedArrayOf<const HBUINT8>& names = StructAfter<const UnsizedArrayOf<const HBUINT8>> (glyphNameIndex);
+    unsigned num = max_index - 257;
+    unsigned length = 0;
+
+    for (unsigned i = 0; i < num; i++)
+    {
+      index_map.set (i+258, length);
+      unsigned name_length = names[length];
+      length += 1 + name_length;
+    }
+  }
+
+  template<typename Iterator,
+           hb_requires (hb_is_iterator (Iterator))>
+  bool serialize (hb_serialize_context_t *c,
+                  Iterator it,
+                  const hb_map_t& index_map) const
+  {
+    TRACE_SERIALIZE (this);
+    auto *out = c->start_embed (this);
+    if (unlikely (!c->check_success (out))) return_trace (false);
+    if (!out->glyphNameIndex.serialize (c, + it
+                                           | hb_map (hb_second)))
+        return_trace (false);
+
+    const UnsizedArrayOf<const HBUINT8> &names = StructAfter<const UnsizedArrayOf<const HBUINT8>> (glyphNameIndex);
+    hb_set_t copied_indices;
+    for (const auto& i : + it | hb_map (hb_first))
+    {
+      if (i < 258) continue;
+      if (!index_map.has (i)) return_trace (false);
+      
+      unsigned name_index = index_map.get (i);
+      if (copied_indices.has (name_index)) continue;
+      copied_indices.add (name_index);
+
+      unsigned name_length = names[name_index] + 1;
+      auto *o = c->allocate_size<HBUINT8> (HBUINT8::static_size * name_length);
+      if (unlikely (!o)) return_trace (false);
+      memcpy (o, &(names[name_index]), HBUINT8::static_size * name_length);
+    }
+
+    return_trace (true);
+  }
+
+  bool subset (hb_subset_context_t *c) const
+  {
+    TRACE_SUBSET (this);
+
+    const hb_map_t &reverse_glyph_map = *c->plan->reverse_glyph_map;
+    hb_map_t old_new_index_map;
+    unsigned i = 0;
+
+    auto index_iter =
```

Comment:
Re-using index_iter will be problematic since it modifies the local variable 'i'. Each time the iterator is run i will be incremented further.

I suggest:
- Moving this iteration code into a for loop that builds the map and computes the max old index.
- Use old_new_index_map.iter() to provide the iterator for serialize.